src/main/java/com/jcabi/ssh/SSHByPassword.java

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright (c) 2014-2015, jcabi.com
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met: 1) Redistributions of source code must retain the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer. 2) Redistributions in binary form must reproduce the above
+ * copyright notice, this list of conditions and the following
+ * disclaimer in the documentation and/or other materials provided
+ * with the distribution. 3) Neither the name of the jcabi.com nor
+ * the names of its contributors may be used to endorse or promote
+ * products derived from this software without specific prior written
+ * permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT
+ * NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
+ * FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+ * THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT,
+ * INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION)
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+ * STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package com.jcabi.ssh;
+
+import com.jcabi.aspects.RetryOnFailure;
+import com.jcabi.aspects.Tv;
+import com.jcabi.log.Logger;
+import com.jcraft.jsch.JSch;
+import com.jcraft.jsch.JSchException;
+import com.jcraft.jsch.Session;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.UnknownHostException;
+import java.util.concurrent.TimeUnit;
+import lombok.EqualsAndHashCode;
+import lombok.ToString;
+import org.apache.commons.lang3.Validate;
+
+/**
+ * SSH channel with authentication by password.
+ * @author Georgy Vlasov (suseika@tendiwa.org)
+ * @version $Id$
+ * @since 1.4
+ * @see SSH For SSH channel with authenticaton using private key.
+ */
+@ToString
+@EqualsAndHashCode(of = { "addr", "port", "login", "password" })
+public final class SSHByPassword implements Shell {
```

Comment:
@darkled Ok, my bad, we don't have to talk to an sshd process. But there is
nothing to mock either, because all SSH and SSHByPassword do is they create
objects you'd need to mock in order to test if these two classes can talk
via SSH.

Now, to test SSHByPassword we can do the same as in SSHTest: use Apache's
Java SSH server implementation. But extracting that tightly coupled code
from SSHTest and rewriting it to accept auth by password would definitely
be a separate task.

On Mon, Jun 22, 2015 at 4:28 PM, Alexey Saenko <notifications@github.com>
wrote:

> In src/main/java/com/jcabi/ssh/SSHByPassword.java
> <https://github.com/jcabi/jcabi-ssh/pull/25#discussion_r32924784>:
>
> > +import java.net.UnknownHostException;
> > +import java.util.concurrent.TimeUnit;
> > +import lombok.EqualsAndHashCode;
> > +import lombok.ToString;
> > +import org.apache.commons.lang3.Validate;
> > +
> > +/**
> > + * SSH channel with authentication by password.
> > + * @author Georgy Vlasov (suseika@tendiwa.org)
> > + * @version $Id$
> > + * @since 1.4
> > + * @see SSH For SSH channel with authenticaton using private key.
> > + */
> > +@ToString
> > +@EqualsAndHashCode(of = { "addr", "port", "login", "password" })
> > +public final class SSHByPassword implements Shell {
>
> @Suseika <https://github.com/Suseika> well, I don't see any sshd string,
> but there are java classes which seems to be mockable :) Do you mean class
> SSHD ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jcabi/jcabi-ssh/pull/25/files#r32924784>.
>